src/privileged/multipreffer/api.js

```diff
@@ -78,22 +78,25 @@ this.FirefoxHooks = {
   },
 
   onUninstalling(addon) {
-    this.handleDisableOrUninstall(addon);
+    this.handleDisableOrUninstall(addon, { uninstall: true });
   },
 
   onDisabled(addon) {
-    this.handleDisableOrUninstall(addon);
+    this.handleDisableOrUninstall(addon, { uninstall: false });
   },
 
-  async handleDisableOrUninstall(addon) {
+  async handleDisableOrUninstall(addon, { uninstall=false }={}) {
```

Comment:
Do we need an object for this param? Can we just pass a boolean directly?

Alternatively, we can avoid needing a parameter by simply moving the `addon.uninstall()` and the comment into the end of the `onUninstalling` function.

Reply:
I try and avoid direct boolean parameters, since callers rarely have enough context to make it clear what the boolean means. Using the object style makes caller code more readable.

That being said, moving the unininstall step to `onUninstalling` is cleaner. I'll do that.